cluster/manager.go

```diff
@@ -867,15 +869,15 @@ func (c *ClusterManager) Start() error {
 	// Start heartbeating to other nodes.
 	go c.startHeartBeat(clusterInfo)
 
```

Comment:
After this point cluster info becomes stale, to disallow others from using it, probably we should move start heart beat and start listeners into a single function.